src/signTxOutput.c

```diff
@@ -384,6 +405,13 @@ static void signTxOutput_handleAssetGroupAPDU(uint8_t* wireDataBuffer, size_t wi
 
 		VALIDATE(view_remainingSize(&view) >= MINTING_POLICY_ID_SIZE, ERR_INVALID_DATA);
 		STATIC_ASSERT(SIZEOF(tokenGroup->policyId) == MINTING_POLICY_ID_SIZE, "wrong policy id size");
+		PRINTF("scrp 1 >%.*H<\n", MINTING_POLICY_ID_SIZE, wireDataBuffer);
+		PRINTF("scrp 2 >%.*H<\n", MINTING_POLICY_ID_SIZE, tokenGroup->policyId);
+		VALIDATE(signTxOutput_lexiCompareViews(
+			make_read_view(tokenGroup->policyId, tokenGroup->policyId + MINTING_POLICY_ID_SIZE),
+			make_read_view(wireDataBuffer, wireDataBuffer + MINTING_POLICY_ID_SIZE)
```

Comment:
I don't like working with `wireDataBuffer` directly, it duplicates the parsing logic (if APDU data format is changed, we will need to modify two different places) and it is not immediately obvious what we do here. Let's consider an alternative:
creating a local buffer where we copy the policyId, then validate it (i.e. compare with previous), and then memcpy it to its proper place. What do you think? (maybe a one-line comment explaining that policyIds have to be ordered properly in the map in tx body)

Reply:
True, that reduces the code duplication, but increases the stack size.

Reply:
It's possible to wrap it in a noinline_due_to_stack function, say, `_parseTokenGroupPolicyId`.